docs/npq_contracts.md

```diff
@@ -0,0 +1,98 @@
+# NPQ Contracts
+
+NPQ Contract holds values that are used to calculate monthly financial statements for a lead provider and cohort.
+
+## Schema Design
+
+```mermaid
+erDiagram
+    Statement }|--|| LeadProvider : ""
+    Statement }|--|| Cohort : ""
+
+    Contract }|--|| Statement : ""
+    Contract }|--|| Course : ""
+    Contract }|--|| ContractTemplate : ""
+
+    LeadProvider {
+        integer id
+        string name
+    }
+
+    Course {
+        integer id
+        string identifier
+    }
+
+    Cohort {
+        integer id
+        integer start_year
+    }
+
+    Statement {
+        integer id
+        integer cohort_id
+        integer lead_provider_id
+        integer month
+        integer year
+        uuid ecf_id
+    }
+
+    Contract {
+        integer id
+        integer statement_id
+        integer course_id
+        integer contract_template_id
+    }
+
+    ContractTemplate {
+        integer id
+        integer recruitment_target
+        integer service_fee_installments
+        integer service_fee_percentage
+        decimal per_participant
+        integer number_of_payment_periods
+        integer output_payment_percentage
+        decimal monthly_service_fee
+        decimal targeted_delivery_funding_per_participant
+        boolean special_course
+        uuid ecf_id
+    }
+```
+
+A `Statement` has many `Contract`, one for each `Course`.
+
+`Contract` belongs to a `ContractTemplate`, which holds contract values
+for statement calculation.
+
```

Comment:
is it worth including a section that describes how contracts span certain timeframes and why we need to maintain older templates while linking the new statements to the newer/different templates?